include/functional/and_then.hpp

```diff
@@ -27,30 +28,25 @@ constexpr inline struct and_then_t final {
 struct and_then_t::apply final {
   static auto operator()(some_expected auto &&v, auto &&fn) noexcept
       -> decltype(auto)
-    requires std::invocable<decltype(fn),
-                            decltype(std::forward<decltype(v)>(v).value())>
+    requires std::invocable<decltype(fn), decltype(FWD(v).value())>
              && (!std::is_void_v<decltype(v.value())>)
```

Comment:
I have a feeling you might be better off if you made a "some_void_expected" concept...